low-level/framework/utils/salt_util.py

```diff
@@ -14,83 +14,51 @@
+import subprocess
+import sys
 
 class SaltInterface:
     """
     Interface class to get common config from common pillar using
     Provisioner API
     """
-
-    __instance = None
-    SALT_FILE = '/etc/salt/minion_id'
-
     CLUSTER_KEY = 'cluster'
     CLUSTER_TYPE_KEY = 'type'
     SSPL_KEY = 'sspl'
     DATASTORE_KEY = 'DATASTORE'
+    SYSTEM_INFORMATION_KEY = 'SYSTEM_INFORMATION'
+    MINION_ID_KEY = 'salt_minion_id'
+    CONSUL_HOST_KEY = 'consul_host'
+    CONSUL_PORT_KEY = 'consul_port'
 
     def __init__(self):
         '''init method'''
-        if SaltInterface.__instance is None:
-            self.utility = Utility()
-            self.pillar_info = None
-            self.node_id = None or self.get_node_id()
-            _result, _err, _ret_code = \
-                    self.utility.execute_cmd(['sudo', 'provisioner', 'pillar_get'])
-            _err = _err.decode("utf-8").rstrip()
-            if _ret_code == 0 and _err == '':
-                self.pillar_info = eval(_result.decode("utf-8").rstrip())
-            self._is_single_node = None or self._is_server_single(_err)
-            self.consul_host = None or self.get_consul_vip(_err)
-            self.consul_port = None or self.get_consul_port(_err)
-            SaltInterface.__instance = self
-
-    @staticmethod
-    def get_singleton_instance():
-        '''Returns an instance of this class'''
-        if SaltInterface.__instance is None:
-            SaltInterface()
-        return SaltInterface.__instance
-
-    @staticmethod
-    def get_node_id():
+        self.node_id = self.get_node_id()
+        self.pillar_storage = PillarStorage()
+        self._is_single_node = self._is_server_single()
+        self.consul_host = self.get_consul_vip()
+        self.consul_port = self.get_consul_port()
+        self.config = configparser.ConfigParser()
+
+    def get_node_id(self):
         '''Returns salt minion_id using salt config file'''
 
         _node_id = 'srvnode-1'
         try:
-            with open(SaltInterface.SALT_FILE, 'r') as salt_file:
-                _node_id = salt_file.read().rstrip()
-            if _node_id is None or _node_id == '':
-                _node_id = 'srvnode-1'
-            logger.info(f'salt_util, Server minion ID: {_node_id}')
-        except OSError as err:
-            if err.errno == errno.ENOENT:
-                logger.error(
-                    f'Problem occured while reading from salt file. \
-                     File path: {SaltInterface.SALT_FILE} does not exist')
-            elif err == errno.EACCES:
-                logger.error(
-                    f'Problem occured while reading from salt file. \
-                     Permission issue while reading from: {SaltInterface.SALT_FILE}')
+            CMD = ['sudo', 'salt-call', '--local', 'grains.get', 'id', '--output=newline_values_only']
+            _result, _err, _ret_code = self.execute_cmd(CMD)
+            if _ret_code == 0 and _err.decode('utf-8') == '':
```

Comment:
we need to start reading 'error' only if return code is non-zero and then check specific things in error string if needed. untangle these checks and depend on return code mainly

Reply:
Done